### ISSUE TITLE
[Suggestion] Improve  asynchronous calls linting

### ISSUE THREAD

**Issue by vladimiry** [mullvad/mullvadvpn-app#924]
I'd suggest enabling the following tslint rules:
```
    "await-promise": true,
    "no-floating-promises": true,
```

If I do it gives me the following output:
```
mullvadvpn-app/gui/src/main/daemon-rpc.ts:359:5
ERROR: 359:5   no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/main/index.ts:402:5
ERROR: 402:5   no-floating-promises  Promises must be handled appropriately
ERROR: 514:5   no-floating-promises  Promises must be handled appropriately
ERROR: 532:7   no-floating-promises  Promises must be handled appropriately
ERROR: 699:7   no-floating-promises  Promises must be handled appropriately
ERROR: 869:7   no-floating-promises  Promises must be handled appropriately
ERROR: 957:7   no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/app.tsx:101:7
ERROR: 101:7   no-floating-promises  Promises must be handled appropriately
ERROR: 177:7   no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/components/Accordion.tsx:59:9
ERROR: 59:9    no-floating-promises  Promises must be handled appropriately
ERROR: 63:7    no-floating-promises  Promises must be handled appropriately
ERROR: 92:29   await-promise         Invalid 'await' of a non-Promise value.
ERROR: 93:27   await-promise         Invalid 'await' of a non-Promise value.

mullvadvpn-app/gui/src/renderer/components/AppButton.tsx:95:5
ERROR: 95:5    no-floating-promises  Promises must be handled appropriately
ERROR: 131:31  await-promise         Invalid 'await' of a non-Promise value.
ERROR: 136:7   no-floating-promises  Promises must be handled appropriately
ERROR: 145:27  await-promise         Invalid 'await' of a non-Promise value.
ERROR: 161:5   no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/components/Connect.tsx:219:11
ERROR: 219:11  await-promise         Invalid 'await' of a non-Promise value.

mullvadvpn-app/gui/src/renderer/components/Login.tsx:75:5
ERROR: 75:5    no-floating-promises  Promises must be handled appropriately
ERROR: 93:5    no-floating-promises  Promises must be handled appropriately
ERROR: 94:5    no-floating-promises  Promises must be handled appropriately
ERROR: 174:20  await-promise         Invalid 'await' of a non-Promise value.
ERROR: 339:5   no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/components/NavigationBar.tsx:386:27
ERROR: 386:27  await-promise         Invalid 'await' of a non-Promise value.
ERROR: 390:26  await-promise         Invalid 'await' of a non-Promise value.

mullvadvpn-app/gui/src/renderer/components/NotificationBanner.tsx:188:9
ERROR: 188:9   no-floating-promises  Promises must be handled appropriately
ERROR: 224:9   no-floating-promises  Promises must be handled appropriately
ERROR: 244:20  await-promise         Invalid 'await' of a non-Promise value.

mullvadvpn-app/gui/src/renderer/containers/AccountPage.tsx:21:7
ERROR: 21:7    no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/containers/LoginPage.tsx:28:7
ERROR: 28:7    no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/containers/PreferencesPage.tsx:33:9
ERROR: 33:9    no-floating-promises  Promises must be handled appropriately
ERROR: 39:7    no-floating-promises  Promises must be handled appropriately

mullvadvpn-app/gui/src/renderer/lib/account-data-cache.ts:40:7
ERROR: 40:7    no-floating-promises  Promises must be handled appropriately
ERROR: 111:7   no-floating-promises  Promises must be handled appropriately
```

**Comment by pronebird** [comment]
Thanks for your suggestion. We definitely should enable these rules to make sure all promises are properly used. However, note that TSLint is going to be deprecated at some point during 2019, so perhaps we would first have to migrate to ESLint.

**Comment by vladimiry** [comment]
Makes total sense. Just wanted to highlight this as a potential issue since in some cases missed `await` on promise call might lead to system instability due to the bugs that are hard to spot and track down.

**Comment by faern** [comment]
Fixed in #1519

**Comment by vladimiry** [comment]
Looks like it's time to close the issue as resolved.
